docs/it/protocolli-di-comunicazione.md

```diff
@@ -0,0 +1,148 @@
+### Lo stack OSI e la distinzione tra TCP e UDP
+
+In base al modello [OSI](https://it.wikipedia.org/wiki/Modello_OSI), i protocolli di comunicazione possono essere divisi in vari livelli ma il concetto di base è il seguente: strutturare la comunicazione in più livelli, dove ogni livello ha un compito specifico e viene definito basandosi sui livelli precedenti. Per esempio, il protocollo HTTP/1.1 è definito al livello 7 e utilizza il protocollo TCP, che è definito invece al livello 4, il quale di basa sul protocollo IP, definito al livello 3. Questo permette di definire protocolli più complessi basandosi su protocolli più semplici e di riutilizzare i protocolli più semplici in più contesti.
+
+Senza però entrare troppo nei dettagli, possiamo dire che i protocolli di comunicazione possono essere divisi in due categorie: quelli orientati alla connessione e quelli non orientati alla connessione.
+
+Il protocollo TCP è un protocollo di comunicazione orientato alla connessione, questo significa che prima di trasmettere le informazioni, il client e il server devono stabilire una connessione tramite un handshake. Questo meccanismo permette di inizializzare la connessione che garantirà che le informazioni vengano trasmesse correttamente e che non vengano perse durante la trasmissione.
+
+Il protocollo UDP è un protocollo di comunicazione non orientato alla connessione, questo significa che non è necessario effettuare meccanismi di inizializzazione per la trasmissione delle informazioni. Vista la mancanza di azioni iniziali come quella dell'handshake, questo tipo di procolli permette di trasmettere le informazioni più velocemente, ma non ne garantisce la correttezza o che esse non vengano perse.
+
+Il modello OSI è una struttura concettuale che ci permette di identificare e catalogare i protocolli di comunicazione, ma solitamente non viene utilizzato direttamente nella pratica. In pratica, i protocolli di comunicazione vengono utilizzati in modo trasparente, senza dover conoscere i dettagli del modello OSI.
+
+### Il protocollo DNS per la risoluzione dei nomi di dominio
+
+Durante la navigazione web, siamo soliti utilizzare i nomi di dominio per accedere ai siti web, come ad esempio `www.google.com`. Questi nomi di dominio sono più facili da ricordare, utilizzare e da condividere. Purtroppo però, i computer non possono utilizzare direttamente i nomi di dominio per connettersi ad un server web: è necessario convertire il nome di dominio in un indirizzo IP. Il protocollo DNS (Domain Name System) è utilizzato per rispondere a questa esigenza.
+
+Il client, prima di effettuare la richiesta ad un server web, deve quindi tradurre il nome di dominio in un indirizzo IP. Per fare questo, il client invia una richiesta ad un server DNS, un server che contiene un database la corrispondenza tra i nomi di dominio e loro indirizzi IP. Solo quando il client riceve dal server DNS l'indirizzo IP corrispondente al nome di dominio, può effettuare la richiesta al server web.
```

Comment:
```suggestion
Il client, prima di effettuare la richiesta ad un server web, deve quindi tradurre il nome di dominio in un indirizzo IP. Per fare questo, il client invia una richiesta ad un server DNS, ossia un server avente nel proprio database la corrispondenza tra i nomi di dominio e loro indirizzi IP. Solo quando il client riceve dal server DNS l'indirizzo IP corrispondente al nome di dominio, può effettuare la richiesta al server web.
```

Reply:
@allevo Cosa ne pensi di questa suggestion? Pensi sia da integrare o preferisci scartarla? Perché nel primo caso la portiamo dentro e procediamo, nel secondo caso procediamo e basta 😀

Rimango in attesa di un tuo feedback 👋